### PR TITLE
Create with-firebase-hosting-and-typescript example

### PR DESCRIPTION
**Pull request by sampsonjoliver**
Adds an example based off of @jthegedus work on firebase hosting, compatible with next v6 and using typescript in both the firebase functions and the next app.

### CODE REVIEW

File: examples/with-firebase-hosting-and-typescript/.firebaserc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "projects": {
+    "default": "retro-e4947"
```
Comment (jthegedus) [inline examples/with-firebase-hosting-and-typescript/.firebaserc]: `<project-id>`

**Review by jthegedus — CHANGES_REQUESTED** [review]

**Review by jthegedus — CHANGES_REQUESTED** [review]

File: examples/with-firebase-hosting-and-typescript/src/app/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "preserveConstEnums": true,
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "lib": [
+      "es6",
+      "dom",
+      "es2016"
+    ],
+    "baseUrl": ".",
+  }
```
[the remark this answers is not in the document]
Comment (sampsonjoliver) [reply examples/with-firebase-hosting-and-typescript/src/app/tsconfig.json]: Very good point, I thought the next typescript plugin might do typechecking so didn't add a `tsc` call.

Have added, and added `noEmit` as well.

**Comment by jthegedus** [comment]
@sampsonjoliver It might also be worth seeing what errors appear when using a custom `_app.js` file as errors arise with the `js` version. Preempting with a note in the readme will save some later debugging. re https://github.com/zeit/next.js/issues/4401#issuecomment-390397914

**Review by jthegedus — APPROVED** [review]
This looks :100:. The only other **suggestions** I would add are:
* use the babel-typescript to also compile the functions, and set `preset-env` to target the cloud functions runtime
* not enforce the next.js repo's lint rules to the tslint rules

Entirely up to you though :+1:

**Review by timneutkens — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**